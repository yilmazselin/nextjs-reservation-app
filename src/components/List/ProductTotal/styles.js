import styled from "styled-components";
import { colors, fontSize, fontWeight, padding } from "../../../utils/styles";

export const Container = styled.div`
  margin-top: 40px;
  margin-bottom: 33px;
`;

export const styles = {
  bigTitleStyle: {
    color: colors.zero,
    fontWeight: fontWeight.bold,
    fontSize: fontSize.lg,
    params: {
      marginBottom: padding.md,
    },
  },
  secondaryTextStyle: {
    color: colors.zero,
    fontWeight: fontWeight.regular,
    fontSize: fontSize.nm,
    params: {
      marginBottom: padding.sm,
    },
  },
  subTitleStyle: {
    color: colors.zero,
    fontSize: fontSize.md,
    fontWeight: fontWeight.bold,
    params: {
      marginTop: padding.md,
    },
  },
};
