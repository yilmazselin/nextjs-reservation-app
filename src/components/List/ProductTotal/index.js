import { useTranslation } from "next-i18next";
import React from "react";
import Text from "../../common/Text";
import { Container, styles } from "./styles";

export default function ProductTotal({ basket }) {
  const { t } = useTranslation("list");
  return (
    <Container>
      <Text {...styles.bigTitleStyle}>{t("list:totalOfProducts")}:</Text>
      <Text data-testid="totalPriceProducts" {...styles.secondaryTextStyle}>
        {t("list:total")}: {basket?.total} {basket?.currency}
      </Text>
      <Text data-testid="taxes" {...styles.secondaryTextStyle}>
        {t("list:taxes")} + {t("list:shipping")}: {basket?.taxes}{" "}
        {basket?.currency}
      </Text>
      <Text {...styles.subTitleStyle}>
        {t("list:generalTotal")}: {basket?.generalTotal} {basket?.currency}
      </Text>
    </Container>
  );
}
