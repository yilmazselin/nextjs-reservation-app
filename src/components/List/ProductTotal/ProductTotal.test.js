import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import "jest-styled-components";
import { useTranslation } from "react-i18next";
import ProductTotal from ".";

jest.mock("react-i18next", () => ({
  useTranslation: jest.fn(),
}));
const tSpy = jest.fn((str) => str);

useTranslation.mockReturnValue({
  t: tSpy,
  i18n: {
    changeLanguage: () => new Promise(() => {}),
  },
});

describe("<ProductTotal />", () => {
  it("Should render correctly", () => {
    const { container } = render(<ProductTotal />);

    expect(container).toMatchSnapshot();
  });

  it("Should viewed the basket data", () => {
    const basket = {
      total: 100,
      taxes: 0,
      generalTotal: 50,
      currency: "TL",
    };
    render(<ProductTotal basket={basket} />);

    expect(screen.getByText(/list:totalOfProducts/)).toBeInTheDocument();
    expect(screen.getByText(/list:total: 100 TL/)).toBeInTheDocument();
    expect(screen.getByTestId("taxes")).toBeInTheDocument();
    expect(screen.getByText(/list:generalTotal: 50 TL/)).toBeInTheDocument();
  });
});
