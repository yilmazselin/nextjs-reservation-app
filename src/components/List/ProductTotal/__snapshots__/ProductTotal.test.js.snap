// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTotal /> Should render correctly 1`] = `
.c1 {
  color: #000000;
  font-size: 24px;
  font-family: "NunitoSans-700","sans-serif";
  margin-bottom: 24px;
}

.c2 {
  color: #000000;
  font-size: 14px;
  font-family: "NunitoSans-500","sans-serif";
  margin-bottom: 16px;
}

.c3 {
  color: #000000;
  font-size: 16px;
  font-family: "NunitoSans-700","sans-serif";
  margin-top: 24px;
}

.c0 {
  margin-top: 40px;
  margin-bottom: 33px;
}

<div>
  <div
    class="c0"
  >
    <p
      class="c1"
      color="#000000"
      font-size="24"
      font-weight="700"
    >
      list:totalOfProducts
      :
    </p>
    <p
      class="c2"
      color="#000000"
      data-testid="totalPriceProducts"
      font-size="14"
      font-weight="500"
    >
      list:total
      : 
       
    </p>
    <p
      class="c2"
      color="#000000"
      data-testid="taxes"
      font-size="14"
      font-weight="500"
    >
      list:taxes
       + 
      list:shipping
      : 
       
    </p>
    <p
      class="c3"
      color="#000000"
      font-size="16"
      font-weight="700"
    >
      list:generalTotal
      : 
       
    </p>
  </div>
</div>
`;
